### DISCUSSION TITLE
Ordinariate Calendars

### DISCUSSION THREAD

**Discussion by shevans02** [romcal/romcal#407] [Ideas]
Is it possible to add the calendars of the Personal Ordinariates? This would be an interesting project. Their additions to the sanctoral calendar could be handled by way of addition to the local calendars library. But the bigger issue is that they do not use Ordinary Time at all. Instead, they have Sundays after Epiphany (first, second, etc.), followed by Pre-Lent (the three Sundays preceding Ash Wednesday, termed Septuagesima, Sexagesima, and Quinquagesima), and then later on, Sundays after Trinity (first, second, etc., all the way till Christ the King). My question is basically whether this temporal difference is able to be integrated into romcal.

**Comment by shevans02** [comment]
I forgot the other major difference in the Temporal Cycle: they have the Octave of Pentecost as well.

**Comment by tukusejssirs** [comment]
> But the bigger issue is that they do not use Ordinary Time at all. Instead, they have Sundays after Epiphany (first, second, etc.), followed by Pre-Lent (the three Sundays preceding Ash Wednesday, termed Septuagesima, Sexagesima, and Quinquagesima), and then later on, Sundays after Trinity (first, second, etc., all the way till Christ the King). My question is basically whether this temporal difference is able to be integrated into romcal.
>
> I forgot the other major difference in the Temporal Cycle: they have the Octave of Pentecost as well.

@shevans02, well, the short answer is _no_, not at the moment at least.

It seems like the liturgical rules of the personal ordinariate you refer to, are similar tp the 1962 calendar revision (of the Roman Rite of the RC Church; aka Extra-Ordinary form) and thus the liturgical rules differ. Thus, unfortunately, in the current form of romcal we cannot simply add celebration definitions of your personal ordinariate. See also issue #23.

While I personally would like see the support of other RC calendars (be it historical revisions of the Roman Rite or the calendars of the personal ordinariates, or even Greek Catholic calendars), we need to refine the Roman Rite calendar output and that is our current priority. Also @emagnier is not really for addition of other (historical) calendar revisions or rites, as neither he nor me know the liturgical rules nor we have the need for it.

In the end, it is possible to add support for other calendars, however, we first need to code the liturgical rules for your personal ordinariate, which is not an easy job. Of course, your liturgical calendar is somewhat similar to the current Roman Rite calendar, however, there are many differences (as you have already noticed) which need to be thought about.

Finally, there are only two active contributors ATM: @emagnier, who is the only coder of romcal, and myself (I analyse stuff, propose new features, manage localisations and partially celebrations inscribed into calendars). Therefore, if you want to help us, we need more contributors, and thus you (and anyone else) are very welcome to join our team. :wink: I have no idea what are your skills or if you even want to help us in any way. If you want to, I think the fastest way is to fork romcal, fix the liturgical rules yourself (ideally with a liturgist of your personal ordinariate, unless it is yourself :wink:). Then we might consider merging your fork into romcal.

**Reply by shevans02** [reply]
Thanks for this helpful response. This all makes sense. My coding abilities are growing but not yet to the point of making this contribution to romcal. If I get to the point of being able, then I would be interested in helping out and I may prepare such a fork. One clarifying note: on the whole, the Ordinariate calendar is closer to the Ordinary Form rather than to the Extraordinary Form (though it shares a couple elements with the latter which I noted); so, it would at least be a smaller project than adding the Extraordinary Form.

**Reply by tukusejssirs** [reply]
> My coding abilities are growing but not yet to the point of making this contribution to romcal.

Any help in any area (not only code contribributions) is warmly welcome! :pray:

> One clarifying note: on the whole, the Ordinariate calendar is closer to the Ordinary Form rather than to the Extraordinary Form (though it shares a couple elements with the latter which I noted); so, it would at least be a smaller project than adding the Extraordinary Form.

I get that, but there are still some differences in the rules.

---

@shevans02, anyway, could you tell us which ordinariate are you interested in please? :pray:

---

@emagnier, anything you want to say about this? :wink:

**Reply by shevans02** [reply]
As to which ordinariates, some projects that I have in mind would want access to all three ordinariate calendars (North American, UK, and Australia). It turns out that Universalis does support all three Ordinariate calendars, and they enable you to import a text readout of the current liturgical date, as well as embed a Google Calendar into your website. So this gives some basic functionality, but romcal would enable more. (Link to US Ordinariate calendar on Universalis: https://universalis.com/usa.ordinariate.thursday/-500/calendar.htm#calendar-select)

**Comment by emagnier** [comment]
Thank you @shevans02 and @tukusejssirs for this discussion. This is an interesting subject.

If I have understood correctly, the sanctoral of the "Personal Ordinariates" is the same as that of the Roman rite. Only the proper of time has some differences. So the idea here would be to be able to "patch" the proper of time of the Roman rite, to include the specific elements from the calendar of the personal Ordinariate. Then any particular calendars (or the general Roman calendar) could be used (based on this redefined proper of time).

I've in mind to introduce a new feature that will allow religious communities to import a "global proper calendar", that will be integrated between the General Roman calendar, and any particular calendar (many religious communities have a proper liturgical calendar that applies for all regions in the world, similarly and in addition to the general Roman calendar). The design of this new feature is still at early stage (in my mind), but technically this isn't too far, and might also address the need described here for the Ordinariate calendars.

**Reply by tukusejssirs** [reply]
Thanks, @emagnier, for sharing your ideas! :pray:

Regarding the ‘religious’ cals, aren’t they applied as a final step after all other (‘standard’) calendars (be it national/diocesan/parish)? Thus celebrations in religious cals override celebrations defined in otner calendars? I might be wrong. 🤔

**Reply by tukusejssirs** [reply]
@emagnier, we could add support of ordinariate cals in a similar way to igneus/calendarium-romanum#49 proposal (related igneus/calendarium-romanum#98). I think we should some additional general config options to romcal to accomplish this.
